packages/sdk/src/client/client.ts

```
@@ -411,6 +414,26 @@ export class Client {
 			method: 'GET',
 		});
 
-		return response.ok;
+		if (!response.ok) {
+			return false;
+		}
+
+		let ok = true;
+		if (response.headers.get('Content-Type') === 'application/json') {
+			const data = (await response.json()) as LogoutResponse;
+			if (data.redirect) {
+				if (options?.redirect) {
+					ok = await options.redirect(data.redirect);
+				} else {
```

This only works on the browser. Is it intended to run this function on the server? At least, I'd check for the existence of the window object.

---

It's the same approach we use with `fetch`, we let it blow up in the server. I will add a check, but what should we do when not in a browser? Throw an error asking the user to pass the redirect function?

---

I'd do nothing. The user can already do something after logout with `after`. The redirect is async anyway.

---

The redirect is required for disconnecting from the auth provider and completing the log out when `logoutOpenidConnectProvider` is true, we need to keep it.

---

I've noticed we don't support `login()` and `logout()` in the server side. `login()` already has a check for it, I'll add the same in `logout()`

---

That's sad because you could issue a logout on the server and redirect yourself.